### ISSUE TITLE
Failed to download stage packages for desktop-qt5 in zesty

### ISSUE THREAD

**Issue by larryprice** [ubuntu/snapcraft-desktop-helpers#62]
When trying to build with the `desktop-qt5` helper on zesty, I get the following error:

```
Error downloading stage packages for part 'desktop-qt5': The package 'appmenu-qt5' was not found.
```

Using `rmadison`, it's clear that appmenu-qt5 is not currently available in zesty. This will likely become a more common issue in the next couple of weeks when zesty is finally released.

**Comment by geekgonecrazy** [comment]
Just ran into this while trying make a few adjustments to the robomongo snap: https://github.com/frankban/robomongo-snap/blob/master/snapcraft.yaml#L16

Any workaround?

**Comment by vranki** [comment]
Zesty has been released, and this is still a blocking issue.

I'm using after: [desktop-qt5] as [desktop/qt5] complains / is deprecated.

**Comment by larryprice** [comment]
Looking through the snapcraft codebase, there doesn't seem to be a way to specify installing from archives that aren't the current release.

I'm beginning to think this part might be abandoned. I see that the tests are only testing the deprecated `desktop/qt5` instead of `desktop-qt5`, and don't actually test the local version (though this may be a limitation of snapcraft). I also cannot build this plugin locally (even removing `appmenu-qt5`) because of an issue with conflicting filesets.

Having said that, I've created a workaround. It's not the prettiest solution, but it should get the job done for now: [https://github.com/larryprice/snapcraft-desktop-qt5](https://github.com/larryprice/snapcraft-desktop-qt5). YMMV, but this makes the demo work for me at the very least.

**Comment by didrocks** [comment]
As you told, there is unfortunately no mechanism in snapcraft to do this per release, and `appmenu-qt5` is still needed in Xenial for instance, which is the base distro version used for them.
Note that you can override parts of a remote part definition (see http://blog.sergiusens.org/posts/The-Snapcraft-Parts-Ecosystem/), which will make you only override the `stage-packages` definition and not the whole logic.

**Comment by droidmonkey** [comment]
The following snippet makes this work on 17.04:

```
parts:
  app:
    source: .
    after: [desktop-qt5]
  desktop-qt5:
    stage-packages:
      - libxkbcommon0
      - ttf-ubuntu-font-family
      - dmz-cursor-theme
      - light-themes
      - shared-mime-info
      - libqt5gui5
      - libgdk-pixbuf2.0-0
      - libqt5svg5 # for loading icon themes which are svg
      - locales-all
```

**Comment by tim-sueberkrueb** [comment]
I think just using an LXC container or `snapcraft cleanbuild` is usally fine. That way you also guarantee every project contributor builds with the same environment.

**Comment by joshirio** [comment]
@droidmonkey Is your 'desktop-qt5' definition a custo one or is it the standard one with just 'appmenu-qt removed'?

I'm trying to build a snap and I ran into the same issue. Any progress or new info?
Same issue on bionic (18.04) and artful (17.10).

**Comment by droidmonkey** [comment]
The original stage packages are found in the desktop-qt5 snapcraft.yaml: https://github.com/Ubuntu/snapcraft-desktop-helpers/blob/master/qt/snapcraft.yaml

**Comment by joshirio** [comment]
@droidmonkey thanks, I saw it but was confused because it was named 'qt5:' instead of 'desktop-qt5'.

BTW found that the development version of snapcraft seems to have fixed it.
Installed it using snapcraft itself (edge channel), as described [here](https://github.com/giowck/symphytum/issues/48#issuecomment-361719120).

**Comment by bitsandsalsa** [comment]
This issue seems to have been fixed by #113.

**Comment by oSoMoN** [comment]
Indeed, thanks for spotting this.
